AudioSpectrumVisualizer.py

```diff
@@ -417,16 +418,20 @@ def full():
 	if(VIDEO == 1):
 		print("Converting image sequence to video.")
 
-		system('ffmpeg -hide_banner -loglevel error -r {} -i "{}/%0d.png" -c:v libx264 -preset ultrafast -crf 16 -y "{}.mp4"'
+		system('ffmpeg -hide_banner -loglevel error -r {} -i "{}/%0d.png" -c:v libx264 -preset ultrafast -crf 16 -pix_fmt yuv420p -y "{}.mp4"'
 			.format(str(FRAMERATE), str(DESTINATION), str(DESTINATION)))
 		
 		processTime = time() - startTime
 		print("Succesfully converted image sequence to video in " + str(format(processTime, ".3f")) + " seconds.")
 
 	if(VIDEOAUDIO == 1):
+		if(args.end == "False"):		#Python reported the END variable as unbound here??? So I had to use args.end to initialise it again
+			END = len(frames)/FRAMERATE
+		else:
+			END = int(args.end)
 		print("Converting image sequence to video (with audio).")
 		
-		system('ffmpeg -hide_banner -loglevel error -r {} -i "{}/%0d.png" -ss {} -i "{}" -t {} -c:v libx264 -preset ultrafast -crf 16 -y "{}.mp4"'
+		system('ffmpeg -hide_banner -loglevel error -r {} -i "{}/%0d.png" -ss {} -i "{}" -t {} -c:v libx264 -preset ultrafast -crf 16 -pix_fmt yuv420p -y "{}.mp4"'
 			.format(str(FRAMERATE), str(DESTINATION), str(START), str(FILENAME), str(END-START), str(DESTINATION)))
```

Comment:
Alternatively we could also just leave out the `-t ...` option when END is "False". No need to calculate length then, that's simpler imo.

We could also leave out the `-t ...` option when START is 0, but that doesn't have a real benefit.

Reply:
Changed this completely by appending the flags in a sort of more 'modular' way, since I also felt like the oroginal ffmpeg call was a bit long. Made those two parts (START and END) conditional.

I'm not quite sure this approach is actually better, since it also makes the code quite a bit longer, and it's not so much more readable as I initially thought...
Your thoughts on this?

Reply:
I think that it's more readable and you removed redundancy that way.

An even better approach would be to use `subprocess.Popen` instead of `os.system`, that way you can provide an array of arguments which should be cleaner and safer (we don't need quotes around filenames that way). This is also the preferred way according to the documentation.

Something like this:
```
# TODO: build array of options
arguments = ['ffmpeg', '-i', filename, 'etc...']

proc = subprocess.Popen(
    arguments,
    stdout=subprocess.PIPE,
    stderr=subprocess.PIPE,
)
# TODO: print ffmpegs output?
proc.wait()
```
Be aware that I did not test this, so it might be wrong.

This also has the benefit that we can [print ffmpegs output](https://stackoverflow.com/a/17413045/4884643).